src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2347,8 +2347,9 @@ public override DeconstructionInfo GetDeconstructionInfo(ForEachVariableStatemen
 
         internal override Symbol RemapSymbolIfNecessaryCore(Symbol symbol)
         {
-            if (!(symbol is LocalSymbol || symbol is ParameterSymbol || symbol is MethodSymbol)
-                || symbol.Locations.IsDefaultOrEmpty)
+            Debug.Assert(symbol is LocalSymbol or ParameterSymbol or MethodSymbol { MethodKind: MethodKind.LambdaMethod });
```

Comment:
>MethodKind.LambdaMethod [](start = 96, length = 23)

It is not clear why the method is necessarily a lambda. Could you please elaborate?

Reply:
The caller ensures it is.

---
In reply to: [550375309](https://github.com/dotnet/roslyn/pull/49953#discussion_r550375309) [](ancestors = 550375309)

Reply:
> The caller ensures it is.

Could you give an example?

---
In reply to: [550380289](https://github.com/dotnet/roslyn/pull/49953#discussion_r550380289) [](ancestors = 550380289,550375309)